autogen/main/cluster.tf.tmpl

```diff
@@ -520,6 +520,10 @@ resource "google_container_node_pool" "pools" {
     }
 
     boot_disk_kms_key = lookup(each.value, "boot_disk_kms_key", "")
+
+    kubelet_config {
```

Comment:
Does this cause nodes to be cordoned and drained when introducing `kubelet_config` with a default of `none`?
If that is the case, it may make sense to make this a dynamic block that is only created if kubelet_config is explicitly set.

Reply:
Yes it will replace the nodes. Recommend to use dynamic block

Reply:
somethink like this should work

```hcl
    dynamic "kubelet_config" {
      for_each = contains(keys(each.value), "cpu_manager_policy") ? [1] : []
      content {
        cpu_manager_policy = lookup(each.value, "cpu_manager_policy")
      }
    }
```

Reply:
This is a good example to adapt here to be more consistent: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/5ba88a6cafc6d99680c02086b50bbe0e5fe989e6/autogen/main/cluster.tf.tmpl#L65

Reply:
@flecno that approach lgtm. Regarding beta-cluster gating, this is already within the conditional.
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/5ba88a6cafc6d99680c02086b50bbe0e5fe989e6/autogen/main/cluster.tf.tmpl#L513